jetty-server/src/main/java/org/eclipse/jetty/server/Server.java

```diff
@@ -368,96 +374,107 @@ public HttpField getDateField()
+
+            // Open network connectors first
+            if (_verifiedStartSequence)
             {
-                try
+                _connectors.stream().filter(NetworkConnector.class::isInstance).map(NetworkConnector.class::cast).forEach(connector->
                 {
-                    connector.open();
-                }
-                catch(Throwable e)
+                    try
+                    {
+                        connector.open();
+                    }
+                    catch(Throwable e)
+                    {
+                        mex.add(e);
+                    }
+                });
+
+                if (mex.size()>0)
                 {
-                    mex.add(e);
+                    // Close any that were open
+                    _connectors.stream().filter(NetworkConnector.class::isInstance).map(NetworkConnector.class::cast).forEach(NetworkConnector::close);
+                    mex.ifExceptionThrow();
                 }
-            });
-            
-            if (mex.size()>0)
-            {
-                // Close any that were open
-                _connectors.stream().filter(NetworkConnector.class::isInstance).map(NetworkConnector.class::cast).forEach(NetworkConnector::close);
-                mex.ifExceptionThrow();
             }
-        }
-        
-        try
-        {
-            // Start the server and components, but #start(LifeCycle) is overridden
-            // so that connectors are not started until after all other components are 
-            // started.
-            super.doStart();
-        }
-        catch(Throwable e)
-        {
-            mex.add(e);
-        }
 
-        // start connectors last
-        for (Connector connector : _connectors)
-        {
+            if (_verifiedStartSequence)
+                mex.ifExceptionThrow();
+
             try
             {
-                connector.start();
+                // Start the server and components, but #start(LifeCycle) is overridden
+                // so that connectors are not started until after all other components are 
+                // started.
+                super.doStart();
             }
             catch(Throwable e)
             {
                 mex.add(e);
             }
-        }
 
-        if (isDumpAfterStart())
-            dumpStdErr();
+            // Throw now if verified start sequence
+            if (_verifiedStartSequence)
+                mex.ifExceptionThrow();
+                
+            // start connectors last
+            for (Connector connector : _connectors)
+            {
+                try
+                {
+                    connector.start();
+                }
+                catch(Throwable e)
+                {
+                    mex.add(e);
+                }
+            }
 
-        mex.ifExceptionThrow();
+            mex.ifExceptionThrow();
+            LOG.info(String.format("Started @%dms",Uptime.getUptime()));
+        }
+        finally
```

Comment:
What happens if something above (not in a try/catch of its own) causes an error?
Such as if ShutdownMonitor.register(this) had an exception?  
This wouldn't be logged / dumped anywhere, right?

Reply:
Any other exception would still be thrown as an exception from the original call to `start()` and the state would end up as `FAILED`. These happen before we open connectors, so that is OK.

However, I think that we should also close any opened connectors if a context fails....